DIPs/DIP1NNN-MVF.md

```diff
@@ -0,0 +1,159 @@
+* [Copyright & License](#copyright--license)
+* [Reviews](#reviews)
+
+## Rationale
+
+At a recent security conference, a Microsoft engineer revealed that approximately 70% of security vulnerabilities in Microsoft products are due to memory safety bugs[2].
+
+As more and more devices become connected to the Internet and other networks, the severity of memory safety exploits increases significantly.  In October of 2016 malicious actors exploited seemingly innoculous devices from printers to baby monitors to perform a distributed denial of service attack on DNS provider Dyn[5].
+
+At the "Trends in Systems Programming" panel at DConf 2017, Walter Bright had this to say[1]:
+
+> I strongly feel that [memory safety] is a coming tsunami in programming languages and we better be ready for it. [...] It's a major issue and I want to be there when the tsunami falls on the beach. [...] It's me looking a couple years ahead, reading the tech news about the kinds of problems people are having and looking "how can we get ahead of this?" so that we're ready when people are going to demand it.  I firmly believe that memory safety will be an absolute requirement moving forward, very soon, for programming language selection. [...] I believe memory safety will kill C.  People are tired of those expensive disasters they have when they have memory corruption bugs,  and malware gets in, and wrecks their system and destroys their customer trust and their products, and they're just not going to put up with it anymore.
+
+With such a strong statement advocating for memory-safety, and trends confirming Walter's prediction, it is alarming that **D is currently NOT memory-safe by default**.
+
+The software development field has been aware of memory-safety issues for decades which has contributed to the rise of more modern general-purpose programming languages (e.g. Go, Rust, C#/VB.Net, Swift, Java) that are memory-safe out-of-the-box.  Those languages require users to employ special techniques to opt out of memory-safety; doing nothing is, efectively, opting in to memory safety.  D, on the other hand, is in the minority in this class of programming languages as **D is currently NOT memory-safe out-of-the-box** and requires users to opt into memory safety using the `@safe` attrbute; doing nothing is, effectively, opting out of memory safety.
+
+As a consequence of the [Default Effect](https://en.wikipedia.org/wiki/Default_effect)[3], users may perceive D as an unsafe language if `@safe` is not the default.
+
+> Defaults might affect the meaning of choice options and thus agents' choices over them. For example, it has been shown that under an opt-in policy in organ donation choosing not to become an organ donor is perceived as a choice of little moral failing. Under an opt-out policy, in contrast, choosing not to be an organ donor is perceived as morally more deficient. These differences in evaluation might affect the rational choice over these options.
```

Comment:
The first quote is clear, but this one and the following are really supposed to be quotes? Quoting what? I think it should be clarified.

Reply:
I added a clarification.  I hope it's sufficient.